src/CMakeLists.txt

```diff
@@ -1,6 +1,21 @@
-# Copyright (c) 2023 The Bitcoin Core developers
+# Copyright (c) 2023-present The Bitcoin Core developers
 # Distributed under the MIT software license, see the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# file COPYING or https://opensource.org/license/mit/.
+
+#=============================
+# secp256k1 subtree
+#=============================
+set(SECP256K1_DISABLE_SHARED ON CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_ECDH OFF CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_RECOVERY ON CACHE BOOL "" FORCE)
+set(SECP256K1_ECMULT_GEN_KB 86 CACHE BOOL "" FORCE)
+include(GetTargetInterface)
+get_target_interface(core_c_flags "" sanitize_interface COMPILE_OPTIONS)
+set(SECP256K1_LATE_CFLAGS ${core_c_flags} CACHE STRING "" FORCE)
+unset(core_c_flags)
+add_subdirectory(secp256k1 EXCLUDE_FROM_ALL)
+set_target_properties(secp256k1 PROPERTIES EXPORT_COMPILE_COMMANDS OFF)
+string(APPEND CMAKE_C_COMPILE_OBJECT " ${APPEND_CPPFLAGS} ${APPEND_CFLAGS}")
```

Comment:
Not sure if this is the right thing, but I think it's orthogonal to this PR and should not hold it up. I've created https://github.com/hebasto/bitcoin/issues/239 to discuss this.